### PR TITLE
feat: Day 13 -14 / HackerRank

### PR DESCRIPTION
**Pull request by StevBL**
## Features

- **feat**: Day 13 - `Abstract Classes`

### CODE REVIEW

**Comment by StevBL** [comment]
## Features +

- **feat**: Day 13 - `Abstract Classes (JS)`
- **feat**: Day 14 - `Scope (Java)`

File: DaysOfCode/abstractClasses.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  _input = _input.split(new RegExp('\n'));
+  main();
+});
+function readLine() { return _input[_index++]; }
+
+/* *** Ignore above this line. *** */
+
+class Book {
+  constructor(title, author) {
+    if (this.constructor === Book) {
```
Comment (RobertBaron) [inline DaysOfCode/abstractClasses.js:20]: what? you might want to use "instance of"

File: DaysOfCode/abstractClasses.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      throw new TypeError('Do not attempt to directly instantiate an abstract class.');
+    } else {
```
Comment (RobertBaron) [inline DaysOfCode/abstractClasses.js:22]: no need for else, you are throwing.

**Review by RobertBaron — CHANGES_REQUESTED** [review]

File: DaysOfCode/abstractClasses.js (hunk 1 above)
Comment (StevBL) [reply DaysOfCode/abstractClasses.js:20]: My code:

image

**Review by RobertBaron — APPROVED** [review]

**Pull request merged**